### ISSUE TITLE
Read-only group

### ISSUE THREAD

**Issue by macro91** [oxen-io/session-pysogs#136]
Hello,

How can I put that only administrators and moderators can publish, and other users only read ?

Thanks

**Comment by jagerman** [comment]
In the current release (0.3.4) this is not easily exposed and requires manual database changes, which is not recommended.

In the upcoming 0.3.5 release (currently in final testing with release expected in the first half of next week) the `sogs` command-line tool has new flags added that allow configuring this; you would use:

```
sogs --set-perms --rooms someroom --remove-perms w
```
to make the room non-writeable to general users.

You can then either assign moderators (using the CLI or from within Session) who will have access.  You can also grant one or more individual users specific write/upload access without making them moderators using:
```
sogs --set-perms --rooms someroom --add-perms w --users 051234567.... 054891841...
```
so that those specific users (but not general users) can post messages.  (Since 0.3.5 also enables blinding, you can specify blinded ids there instead of session ids if you only know the blinded ids).

(There are various other things you can do with this: for instance you can `--remove-perms w` on specific users to make a room generally writeable but remove the ability for some users to post; upload permissions can be controlled distinctly by setting/removing the `u` permission, and so on).

**Comment by macro91** [comment]
Thank you.

**Comment by jagerman** [comment]
This is available now in 0.3.5.  Closing this.
